### PR TITLE
<ExchangeAccount />

### LINKED ISSUE #89: Implement an <ExchangeAccount /> component
Image

Blocked by:
- [x] #95

### PR DESCRIPTION
**Pull request by vitalii-ilchenko-ssa-group**
Closes #89

### CODE REVIEW

File: packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  PieCustomLayerProps: {},
+  ResponsivePie: ResponsivePieMock,
+}));
+
+describe('ExchangeAccount', () => {
+  it('Renders with platform', () => {
+    const { getByText } = render(
+      <ExchangeAccount
+        {...commonProps}
+        deleteOnClick={jest.fn()}
```
Comment (react-developer-ssa-group) [inline packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx:24]: Could we move `deleteOnClick={jest.fn()}` to the commonProps also? And rewrite this prop when we need it

File: packages/widgets/src/components/ExchangeAccount/ExchangeAccount.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { BalancePieChart, WithLink } from '@components/AccountBalance';
+import { ExchangeAccountProps } from './types';
+import {
+  CardWrapper,
+  HeadWrapper,
+  PlatformWrapper,
+  RemoveButton,
+  StatusWrapper,
+  ChartWrapper,
+} from './styles';
```
Comment (react-developer-ssa-group) [inline packages/widgets/src/components/ExchangeAccount/ExchangeAccount.tsx:19]: I think, we can use the next approach:
- rename each style to the component name, where it uses (CardBase, CardHeader, etc...)
- use `import * as S from './styles'`
- use `css={S.CardBase}` in the codebase

What do you think about this approach?

File: packages/widgets/src/components/ExchangeAccount/styles.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  gap: 9px;
+  margin-right: 13px;
+`;
+
+export const StatusWrapper = (theme: Theme) => css`
+  width: 100%;
+  text-align: center;
+  padding: 4px;
+
+  &.Active {
```
Comment (react-developer-ssa-group) [inline packages/widgets/src/components/ExchangeAccount/styles.ts:68]: I think, it's better to use `active` class name

File: packages/widgets/src/components/ExchangeAccount/styles.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    color: ${theme.colors.greenDark};
+    background: ${theme.colors.greenMint};
+  }
+
+  &.NotAvailable {
```
Comment (react-developer-ssa-group) [inline packages/widgets/src/components/ExchangeAccount/styles.ts:73]: I think, it's better to use `not-available` class name

File: packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    expect(mockOnClick).toBeCalledTimes(1);
+  });
+
+  it('Clicks on card', async () => {
+    const user = userEvent.setup();
+    const mockOnClick = jest.fn();
+    const { getByTestId } = render(
+      <ExchangeAccount
+        {...commonProps}
+        deleteOnClick={jest.fn()}
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx:99]: Check that this handler was not called when the card was clicked

File: packages/widgets/src/components/ExchangeAccount/types.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { AccountBalanceProps } from '@components/AccountBalance';
+
+export interface ExchangeAccountProps {
+  platform: string | JSX.Element;
+  title: string;
+  status: 'Active' | 'NotAvailable';
+  onClick?: () => void;
+  deleteOnClick: () => void;
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/ExchangeAccount/types.ts:8]: I would call it `onDelete` or `onRemove`

File: packages/widgets/src/components/ExchangeAccount/types.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  link?: string;
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/ExchangeAccount/types.ts:9]: It should be `To`.

**Review by arthur-kashaiev-ssa-group — CHANGES_REQUESTED** [review]

File: packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx (hunk 1 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx:24]: Yes, you are right. Done.

File: packages/widgets/src/components/ExchangeAccount/ExchangeAccount.tsx (hunk 2 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccount/ExchangeAccount.tsx:19]: Yes, that's a much better, good approach. Done

File: packages/widgets/src/components/ExchangeAccount/styles.ts (hunk 3 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccount/styles.ts:68]: Done.

File: packages/widgets/src/components/ExchangeAccount/styles.ts (hunk 4 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccount/styles.ts:73]: Done.

File: packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx (hunk 5 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccount/ExchangeAccount.spec.tsx:99]: Done.

File: packages/widgets/src/components/ExchangeAccount/types.ts (hunk 6 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccount/types.ts:8]: Done.

File: packages/widgets/src/components/ExchangeAccount/types.ts (hunk 7 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccount/types.ts:9]: Done.

**Review by arthur-kashaiev-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 12       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 4  | ❗       |
    


<details>
<summary>🟡 storybook/widgets-exchange-account--default__[w1440px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--default__[w1920px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--default__[w390px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--default__[w900px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--list__[w1440px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--list__[w1920px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--list__[w390px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--list__[w900px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--with-link__[w1440px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--with-link__[w1920px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--with-link__[w390px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/widgets-exchange-account--with-link__[w900px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



*❗ custom/widgets-navbar-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-progressInfo__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-progressInfo__[w900px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-user-profile--is-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ storybook/utility-responsive-image--responsive-image__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Review by react-developer-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 6  | ❗       |
    


<details>
<summary>❗ custom/widgets-progressInfo__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-progressInfo__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-user-profile--is-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-exchange-account--list__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-user-profile--default__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-user-profile--default__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 2  | ❗       |
    


*❗ storybook/components-indicator--with-content__[w900px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/utility-responsive-image--responsive-image__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 5  | ❗       |
    


*❗ custom/widgets-navbar-opened__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-navbar-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-progressInfo__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-exchange-account--list__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-heart-rate--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Pull request merged**